### PR TITLE
[pcsc] Fix flaky shutdown

### PR DESCRIPTION
**Pull request by emaxx-google**
Fix flaky memory leaks and drop magic sleep in the PC/SC shutdown logic, in favor of deterministic signaling and waiting.

### CODE REVIEW

**Review by ViktoriiaKovalova — APPROVED** [review]

File: third_party/pcsc-lite/src/src/hotplug_libusb.c [hunk 1]
```diff
@@ -519,24 +495,23 @@ static void * HPEstablishUSBNotifications(int pipefd[2])
 		do_polling = true;
 	}
 
+	char dummy;
```
Comment (ViktoriiaKovalova) [inline third_party/pcsc-lite/src/src/hotplug_libusb.c:498]: this can be moved to `else` clause?

Comment (emaxx-google) [reply third_party/pcsc-lite/src/src/hotplug_libusb.c:498]: We also use it below, in `write()`... But maybe I should try refactoring this into smaller functions.

Comment (ViktoriiaKovalova) [reply third_party/pcsc-lite/src/src/hotplug_libusb.c:498]: ack

**Comment by emaxx-google** [comment]
Didn't manage to repro it, but I have one theory... Let's see if commit#2 passes the bots.

**Comment by emaxx-google** [comment]
After stress-testing this for hours, I managed to catch some crashes, but they seem to be orthogonal. So, given the huge benefit of this PR for the developer experience, and given that it should only affect behavior in tests, let's land it for now.

**Pull request merged**